screens/TournamentDetails.tsx

```diff
@@ -9,144 +9,165 @@ import { Text, View, LinkButton, TextInput } from "../components/Themed";
 import { AuthUserContext } from "../contexts/AuthContext";
 import { Tournament, Bracket } from "../models";
 import { ScreenWithNavigation } from "../types";
+import ButtonStyles from "../components/ButtonStyles";
 
-type DetailScreen = ScreenWithNavigation & { route: Route<any> }
+type DetailScreen = ScreenWithNavigation & { route: Route<any> };
 
-export default function TournamentDetailsScreen({ navigation, route }: DetailScreen) {
-
-    const { user } = useContext(AuthUserContext);
-    const { id: tournamentId } = route.params as { id: string };
-
-    // TODO: Update data and send to API when done
-    const [tournament, setTournament] = useState<Tournament | undefined>();
-    const [loading, setLoading] = useState(true);
-
-    const [bracket, setBracket] = useState<Bracket>({name: ''})
-
-    // For Adding brackets
-    const [showAddBracket, setShowAddBracketModal] = useState(false);
-
-    const fetchTournamentDetails = async () => {
-        const data = await getTournament(tournamentId, user);
-        setTournament(data);
-        setLoading(false)
-    }
+export default function TournamentDetailsScreen({
+  navigation,
+  route,
+}: DetailScreen) {
+  const { user } = useContext(AuthUserContext);
+  if (!user) {
+    throw new Error("User not defined");
+  }
```

Comment:
Do we actually need any of these `!user` checks - since we do the check to decide on what stack to render in the navigation here: https://github.com/probicheaux/ReactBracket/blob/main/navigation/index.tsx#L61

Reply:
So the reason I wanted to do that is because the type of user in the context is `User | null` because I have to give a default of `null`. So this fixes the typing so that the `user` is of type `User` strictly. Is this the way to do that or would `user as User` work, and is that possible?